### ISSUE TITLE
File upload question doesn't do server-side validation of file format type

### ISSUE THREAD

**Issue by caitlinshk** [civiform/civiform#6301]
**Describe the bug**
Our file upload question restricts the allowed file types to be any image (jpg, png, etc.) or a PDF. When you click "Choose File", the file picker on my Mac only shows images and PDFs as available for selection. However, if I manually change the file picker to be "All Files" instead of "Custom Files":

![image](https://github.com/civiform/civiform/assets/147177873/d7ee9fe5-5cd2-4bdd-b3f6-3067c519a6ca)

Then I'm able to upload any kind of file I want (.zip, .pkg, anything) and get it successfully uploaded to the deployment's bucket.

Another way to expose this problem is if you manually edit the HTML to remove the `"accept="image/*,.pdf"` from the `<input type="file">`.

I was able to reproduce this on Seattle's deployment.

I think this may be a possible attack vector for Program Admins? Program Admins will need to download applicant files to verify the application. If we allow any type of file to be uploaded by an applicant, then Program Admins could download potentially harmful files.

**To Reproduce**
1. An an applicant, start applying to a program that requires you to upload a file.
2. Go to the file upload question.
3. Click "Choose File".
4. In the file picker, change it to allow "All Files" selected instead of "Custom Files".
5. Choose a file that's not an image or PDF.
6. Click Save and Next.
7. **Unexpected**: The file is successfully saved.

**Expected behavior**
I'd expect an error to be thrown saying the file is an invalid type and that the file isn't uploaded to the deployment's bucket.

**Screenshots**
Here's a screenshot of my "Review" page as an applicant showing my succesfully uploaded .zip file. (And if I click the text, the file does successfully download.)

![image](https://github.com/civiform/civiform/assets/147177873/8a355710-1517-43fc-8df4-98c9132179d9)

**Comment by gwendolyngoetz** [comment]
Duplicate of #5218

**Comment by gwendolyngoetz** [comment]
We should do this in the application ourselves and not rely on AWS specific tooling. Otherwise we'll have to do it for AWS, Azure, etc, and self-hosted all separately.

**Comment by caitlinshk** [comment]
True, though I think we'll want to do it in AWS as well. When a file is uploaded, the actual file is only sent to AWS; our application only receives the name of the file once the file is successfully uploaded. We could still check on our side that the extension matches our allowed types (though I think it's possible to disguise an `.exe` file as `something.png`?) and not save the file key if the extension doesn't match, but that's the most we can do and the file would still be present in our AWS bucket. Using AWS to enforce this will ensure the file is never uploaded to the bucket in the first place.

**Comment by gwendolyngoetz** [comment]
Ugh. It does? Yuck. We should discuss the requirements on this more before whoever takes this on. The file types should be configurable via the deployment. For example say Seattle wanted to allow a PDF and Arkansas didn't.

**Comment by caitlinshk** [comment]
Agreed, we can discuss more! I could also see it being on a per-question basis instead of per-deployment basis (though maybe that complexity isn't worthwhile).

**Comment by rockycodes** [comment]
@tallulahkay will this be addressed with your changes?

**Comment by tallulahkay** [comment]
@rockycodes Yes!
